### PR TITLE
feat: Add orchestration of settlement reports.

### PR DESCRIPTION
**Pull request by michaelmajgaard**
# Description

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)
- [ ] Documentation has been updated
- [ ] The integration [event catalog](https://energinet.atlassian.net/wiki/spaces/D3/pages/555581556/Event+catalog) has been updated
- [x] C4 diagrams have been updated and Team Outlaws has been informed about relevant changes

### CODE REVIEW

File: source/dotnet/.editorconfig [hunk 1]
```diff
@@ -530,7 +530,15 @@ dotnet_diagnostic.SX1101.severity = error
 dotnet_diagnostic.SX1309.severity = error
 
 ##############################################################
-# C# Test Conventions (overrides default configuration)       #
+# Durable Functions Orchestrator                             #
+##############################################################
+[**/Orchestrations/Functions/**/*Orchestrator.cs]
```
Comment (defectiveAi) [inline source/dotnet/.editorconfig:535]: Rename to Orchestration.

**Review by dstenroejl — COMMENTED** [review]
Added review comments

File: source/dotnet/database-migration/Scripts/202405081100_Add_SettlementReport_table.sql
[the remark this answers is not in the document]
Comment (defectiveAi) [reply source/dotnet/database-migration/Scripts/202405081100_Add_SettlementReport_table.sql:1]: Currently, the reports are to be deleted after only a week. We expect the list of reports to be extremly short.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/Persistence/IDatabaseContext.cs
[the remark this answers is not in the document]
Comment (defectiveAi) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/Persistence/IDatabaseContext.cs:1]: Context renamed.

File: source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/Activities/FinalizeSettlementReportActivity.cs
[the remark this answers is not in the document]
Comment (defectiveAi) [reply source/dotnet/wholesale-api/Orchestrations/Functions/SettlementReports/Activities/FinalizeSettlementReportActivity.cs:1]: Whoopsie. Fixed.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Application/SettlementReports/SettlementReportFileRequestHandlerIntegrationTests.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+using Energinet.DataHub.Wholesale.CalculationResults.IntegrationTests.Fixtures;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.SettlementReports_v2.Models;
+using Energinet.DataHub.Wholesale.Common.Infrastructure.Options;
+using Microsoft.Extensions.Options;
+using NodaTime;
+using Xunit;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.IntegrationTests.Application.SettlementReports;
+
+public sealed class SettlementReportFileRequestHandlerIntegrationTests : TestBase<SettlementReportFileRequestHandler>,
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Application/SettlementReports/SettlementReportFileRequestHandlerIntegrationTests.cs:31]: Ah, yes. Fixed.

I had some issues debugging code with lazy initializer present, so I used Create instead. But all tests should now use Sut property instead.

**Review by FirestarJes — APPROVED** [review]

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/Extensions/Options/SettlementReportStorageOptions.cs
[the remark this answers is not in the document]
Comment (defectiveAi) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/Extensions/Options/SettlementReportStorageOptions.cs:1]: Done

**Pull request merged**